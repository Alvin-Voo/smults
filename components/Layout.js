import React from 'react';
import {Container} from 'semantic-ui-react';
// import 'semantic-ui-css/semantic.min.css';
import '../css/style.css';
import Header from './Header';
import Footer from './Footer';
import Head from 'next/head';


export default props => {
  return(
    <Container style={{padding: '2em 0em'}}>
      <Head>
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.1/semantic.min.css"></link>
      </Head>
      <Header item={props.item}/>
      {props.children}
      <Footer/>
    </Container>
  )
}
