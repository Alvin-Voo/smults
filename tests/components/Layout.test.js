import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../components/Layout';

const wrapper = shallow(<Layout item="search"><div className="children" /></Layout>);

test('should render Layout with item and children', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should pass in the proper item into Header as prop', () => {
  expect(wrapper.find('Header').prop('item')).toBe('search');
});

test('should have the proper children as specified', () => {
  expect(wrapper.find('.children')).toBeTruthy();
});
