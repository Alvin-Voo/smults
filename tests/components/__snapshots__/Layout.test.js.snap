// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Layout with item and children 1`] = `
<Container
  style={
    Object {
      "padding": "2em 0em",
    }
  }
>
  <Header
    item="search"
  />
  <div
    className="children"
  />
  <Component />
</Container>
`;
